docs/block-client-apis/checkout/checkout-api.md

```diff
@@ -14,6 +18,68 @@ This document gives an overview of some of the major architectural components/AP
     -   [Hooks](#hooks)
         -   [`usePaymentMethodInterface`](#usepaymentmethodinterface)
 
+### Data Stores
+
+We are transitioning much of what is now available in Contexts, to @wordpress/data stores.
+
+#### Checkout Data Store
+
+This is responsible for holding all the data required for the checkout process.
+
+The following data is available:
+
+-   `status`: The status of the current checkout. Possible values are `pristine`, `idle`, `processing`, `complete`, `before_processing` or `after_processing`
+-   `hasError`: This is true when anything in the checkout has created an error condition state. This might be validation errors, request errors, coupon application errors, payment processing errors etc.
+-   `redirectUrl`: The current set url that the checkout will redirect to when it is complete.
+-   `orderId`: The order id for the order attached to the current checkout.
+-   `customerId`: The ID of the customer if the customer has an account, or `0` for guests.
+-   `calculatingCount`: If any of the totals, taxes, shipping, etc need to be calculated, the count will be increased here
+-   `processingResponse`:The result of the payment processing
+-   `useShippingAsBilling`: Should the billing form be hidden and inherit the shipping address?
+-   `shouldCreateAccount`: Should a user account be created with this order?
+-   `extensionData`: Custom checkout data passed to the store API on processing.
```

Comment:
Maybe add that this is used by plugins that extend the Cart and Checkout blocks?